### ISSUE TITLE
com.hazelcast.jet.JobStatusListenerTest.testListener_waitForCompletion[onClient] [HZ-2350]

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast#24182]
_master_ (commit 336728cfb054666fb5e4c4c93f74feaebeec4dc4)

Failed on ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8-nightly/98/testReport/com.hazelcast.jet/JobStatusListenerTest/testListener_waitForCompletion_onClient_/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at com.hazelcast.jet.JobStatusListenerTest.lambda$assertHasNoListenerEventually$13(JobStatusListenerTest.java:264)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1288)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1392)
	at com.hazelcast.jet.JobStatusListenerTest.assertHasNoListenerEventually(JobStatusListenerTest.java:263)
	at com.hazelcast.jet.JobStatusListenerTest.testListenerDeregistration_onCompletion(JobStatusListenerTest.java:259)
	at sun.reflect.GeneratedMethodAccessor440.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:381)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:41)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:825)

```
</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: testLightListener_waitForCompletion[onClient] in 0.218 seconds.
Started Running Test: testListener_waitForCompletion[onClient]
06:41:00,512 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Creating job classLoader for job 09a5-b515-6ac4-0001
06:41:00,512 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Create processor classloader map for job 09a5-b515-6ac4-0001
06:41:00,513 DEBUG || - [Planner] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - Watermarks in the pipeline will be throttled to 1000
06:41:00,513  INFO || - [JobCoordinationService] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Starting job 09a5-b515-6ac4-0001 based on submit request
06:41:00,515  INFO || - [MasterJobContext] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Didn't find any snapshot to restore for job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001
06:41:00,515  INFO || - [MasterJobContext] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Start executing job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001, execution graph in DOT format:
digraph DAG {
	"finiteStream" [localParallelism=1];
	"mapWithUpdatingSink(runCount)" [localParallelism=2];
	"finiteStream" -> "mapWithUpdatingSink(runCount)" [label="partitioned", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
06:41:00,515 DEBUG || - [MasterJobContext] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Building execution plan for job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001
06:41:00,515 DEBUG || - [MasterJobContext] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Built execution plans for job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001
06:41:00,515 DEBUG || - [InitExecutionOperation] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Initializing execution plan for job 09a5-b515-6ac4-0001, execution 09a5-b4fd-b7d6-0001 from [127.0.0.1]:5701
06:41:00,515 TRACE || - [ClientListenerService] hz.client_2014.eventRegistration- - hz.client_2014 [dev] [5.3.0-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.jet.impl.ClientJobProxy$1@dfdec71f, handler=com.hazelcast.jet.impl.ClientJobProxy$ClientJobStatusEventHandler@4cda394a} to MockedClientConnection{localAddress=[127.0.0.1]:40002, super=ClientConnection{alive=true, connectionId=2, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2023-04-03 06:41:00.515, lastWriteTime=2023-04-03 06:41:00.515, closedTime=never}}
06:41:00,516  INFO || - [JobExecutionService] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution plan for jobId=09a5-b515-6ac4-0001, jobName='09a5-b515-6ac4-0001', executionId=09a5-b4fd-b7d6-0001 initialized
06:41:00,516 DEBUG || - [InitExecutionOperation] hz.JobStatusListenerTest_goofy_galileo.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Initializing execution plan for job 09a5-b515-6ac4-0001, execution 09a5-b4fd-b7d6-0001 from [127.0.0.1]:5701
06:41:00,516 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_goofy_galileo.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Creating job classLoader for job 09a5-b515-6ac4-0001
06:41:00,516 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_goofy_galileo.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Create processor classloader map for job 09a5-b515-6ac4-0001
06:41:00,517  INFO || - [JobExecutionService] hz.JobStatusListenerTest_goofy_galileo.cached.thread-12 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Execution plan for jobId=09a5-b515-6ac4-0001, jobName='09a5-b515-6ac4-0001', executionId=09a5-b4fd-b7d6-0001 initialized
06:41:00,517 DEBUG || - [MasterJobContext] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Init of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 was successful
06:41:00,517 DEBUG || - [MasterJobContext] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Executing job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001
06:41:00,517  INFO || - [JobExecutionService] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Start execution of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 from coordinator [127.0.0.1]:5701
06:41:00,517  INFO || - [JobExecutionService] hz.JobStatusListenerTest_goofy_galileo.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Start execution of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 from coordinator [127.0.0.1]:5701
06:41:00,517 DEBUG || - [JobExecutionService] hz.JobStatusListenerTest_focused_galileo.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Completed execution of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001
06:41:00,517 DEBUG || - [JobExecutionService] hz.JobStatusListenerTest_focused_galileo.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 completed
06:41:00,517 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-3 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 received response to StartExecutionOperation from [127.0.0.1]:5701: null
06:41:00,518 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_goofy_galileo.cached.thread-12 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=09a5-b515-6ac4-0001
06:41:00,518 DEBUG || - [JobExecutionService] hz.JobStatusListenerTest_goofy_galileo.cached.thread-12 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Completed execution of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001
06:41:00,518 DEBUG || - [JobExecutionService] hz.JobStatusListenerTest_goofy_galileo.cached.thread-12 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Execution of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 completed
06:41:00,518 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-4 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 received response to StartExecutionOperation from [127.0.0.1]:5702: null
06:41:00,518 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-4 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 was successful
06:41:00,518 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=09a5-b515-6ac4-0001
06:41:00,518  INFO || - [MasterJobContext] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution of job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 completed successfully
	Start time: 2023-04-03T06:41:00.514
	Duration: 00:00:00.004
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
06:41:00,518 DEBUG || - [JobCoordinationService] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] job '09a5-b515-6ac4-0001', execution 09a5-b4fd-b7d6-0001 is completed
06:41:00,519 TRACE || - [ClientListenerService] hz.client_2014.eventRegistration- - hz.client_2014 [dev] [5.3.0-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.jet.impl.ClientJobProxy$1@dfdec71f, handler=com.hazelcast.jet.impl.ClientJobProxy$ClientJobStatusEventHandler@4cda394a} to MockedClientConnection{localAddress=[127.0.0.1]:40002, super=ClientConnection{alive=true, connectionId=2, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2023-04-03 06:41:00.519, lastWriteTime=2023-04-03 06:41:00.515, closedTime=never}}
06:41:00,519 TRACE || - [ClientListenerService] hz.client_2014.eventRegistration- - hz.client_2014 [dev] [5.3.0-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.jet.impl.ClientJobProxy$1@dfdec71f, handler=com.hazelcast.jet.impl.ClientJobProxy$ClientJobStatusEventHandler@4cda394a} to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2023-04-03 06:41:00.509, lastWriteTime=2023-04-03 06:41:00.509, closedTime=never}}
06:41:00,519 TRACE || - [ClientListenerService] hz.client_2014.eventRegistration- - hz.client_2014 [dev] [5.3.0-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.jet.impl.ClientJobProxy$1@dfdec71f, handler=com.hazelcast.jet.impl.ClientJobProxy$ClientJobStatusEventHandler@4cda394a} to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2023-04-03 06:41:00.519, lastWriteTime=2023-04-03 06:41:00.519, closedTime=never}}
06:41:00,519 ERROR || - [ClientListenerService] hz.client_2014.event-14858 - hz.client_2014 [dev] [5.3.0-SNAPSHOT] hz.client_2014.event-14858 caught an exception while processing:com.hazelcast.client.impl.spi.impl.listener.ClientListenerServiceImpl$ClientEventProcessor@64fc2490
java.lang.NullPointerException: null
	at java.util.concurrent.ConcurrentHashMap.replaceNode(ConcurrentHashMap.java:1117) ~[?:1.8.0]
	at java.util.concurrent.ConcurrentHashMap.remove(ConcurrentHashMap.java:1108) ~[?:1.8.0]
	at com.hazelcast.client.impl.spi.impl.listener.ClientListenerServiceImpl.lambda$removeListener$8(ClientListenerServiceImpl.java:355) ~[classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:1.8.0]
	at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (08-17-2022)]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:111) ~[classes/:?]
06:41:00,664 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:05,664 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:10,664 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:15,664 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:20,664 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:25,664 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:30,665 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:35,665 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:40,665 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:45,665 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:50,664 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:41:55,664 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:00,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:05,667 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:10,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:15,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:20,667 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:25,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:30,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:35,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:40,667 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:45,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-12 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:50,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-12 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:42:55,666 DEBUG || - [JobRepository] hz.JobStatusListenerTest_focused_galileo.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
06:43:00,626  INFO |testListener_waitForCompletion[onClient]| - [SimpleTestInClusterSupport] Time-limited test - Ditching 14 jobs in SimpleTestInClusterSupport.@After: [09a5-b509-4f49-0002, 09a5-b509-4f47-0002, 09a5-b509-4f44-0002, 09a5-b509-4f42-0002, 09a5-b509-4f40-0002, 09a5-b509-4f41-0002, 09a5-b4fd-6309-0001, 09a5-b515-6ac0-0001, 09a5-b515-6ac2-0001, 09a5-b515-6ac4-0001, 09a5-b4fd-6303-0001, 09a5-b4fd-6302-0001, 09a5-b4fd-6305-0001, 09a5-b4fd-6304-0001]
06:43:00,637  INFO |testListener_waitForCompletion[onClient]| - [SimpleTestInClusterSupport] Time-limited test - Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/runCount]
BuildInfo right after testListener_waitForCompletion[onClient](com.hazelcast.jet.JobStatusListenerTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20230403', buildNumber=20230403, revision=336728c, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testListener_waitForCompletion[onClient](com.hazelcast.jet.JobStatusListenerTest):'
06:41:00, accumulated pauses: 37 ms, max pause: 1 ms, pauses over 1000 ms: 0
06:41:05, accumulated pauses: 44 ms, max pause: 4 ms, pauses over 1000 ms: 0
06:41:10, accumulated pauses: 36 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:41:15, accumulated pauses: 44 ms, max pause: 2 ms, pauses over 1000 ms: 0
06:41:20, accumulated pauses: 57 ms, max pause: 6 ms, pauses over 1000 ms: 0
06:41:25, accumulated pauses: 54 ms, max pause: 8 ms, pauses over 1000 ms: 0
06:41:30, accumulated pauses: 39 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:41:35, accumulated pauses: 45 ms, max pause: 1 ms, pauses over 1000 ms: 0
06:41:40, accumulated pauses: 43 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:41:45, accumulated pauses: 40 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:41:50, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:41:55, accumulated pauses: 44 ms, max pause: 4 ms, pauses over 1000 ms: 0
06:42:00, accumulated pauses: 46 ms, max pause: 2 ms, pauses over 1000 ms: 0
06:42:05, accumulated pauses: 40 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:42:10, accumulated pauses: 42 ms, max pause: 2 ms, pauses over 1000 ms: 0
06:42:15, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:42:20, accumulated pauses: 41 ms, max pause: 2 ms, pauses over 1000 ms: 0
06:42:25, accumulated pauses: 38 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:42:30, accumulated pauses: 41 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:42:35, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:42:40, accumulated pauses: 42 ms, max pause: 1 ms, pauses over 1000 ms: 0
06:42:45, accumulated pauses: 39 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:42:50, accumulated pauses: 41 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:42:55, accumulated pauses: 39 ms, max pause: 0 ms, pauses over 1000 ms: 0
06:43:00, accumulated pauses: 4 ms, max pause: 0 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-IbmJDK8-nightly/98/

**Comment by frant-hartm** [comment]
@k-jamroz I think the SQL team made some changes in this area. Could you guys have a look?

**Comment by k-jamroz** [comment]
@burakgok can you have a look at it?

**Comment by burakgok** [comment]
It seems that the job is completed before the listener is registered. This is expected to be resolved by #24035 [[HZ-2182]].
